Backend/Tumultu/Infrastructure/DataProviders/Memory/Write/InMemoryRepository.cs

```diff
@@ -0,0 +1,57 @@
+using Ardalis.GuardClauses;
+using Tumultu.Application.Common.Interfaces;
+using Tumultu.Domain.Common;
+
+namespace Tumultu.Infrastructure.DataProviders.Memory.Write;
+
+public abstract class InMemoryRepository<TEntity,TId> : IRepository<TEntity,TId> where TEntity: BaseEntity<TId>
+{
+    public Task<IEnumerable<TEntity>> GetAllAsync()
```

Comment:
Once again as previously, I wonder if we should have queries in this type of repositories. This is to be discussed.